Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm

```diff
@@ -668,6 +675,135 @@
 }
 #endif
 
+Node* WebPage::clickableNodeAtLocation(const FloatPoint& viewportLocation, FloatPoint& adjustedViewportLocation) const
+{
+#if PLATFORM(IOS_FAMILY)
+    return Ref(m_page->mainFrame())->nodeRespondingToClickEvents(viewportLocation, adjustedViewportLocation);
+#else
+    UNUSED_PARAM(adjustedViewportLocation);
+
+    Ref frame = CheckedRef(m_page->focusController())->focusedOrMainFrame();
+    RefPtr frameView = frame->view();
+    if (!frameView)
+        return nullptr;
+
+    auto center = frameView->windowToContents(roundedIntPoint(viewportLocation));
```

Comment:
`pointInContentsCoordinates`? I don't think the point is not necessarily the center of something.